### PR TITLE
Add dhwio data layout

### PR DESCRIPTION
**Pull request by tkarna**
More updates for tensorflow integration

- api: add dhwio data layout
- fix fwd pooling: initialize src format

### CODE REVIEW

File: src/cpu/jit_uni_pooling.hpp [hunk 1]
```diff
@@ -86,6 +86,8 @@ struct jit_uni_pooling_fwd_t: public cpu_primitive_t {
 
     protected:
         virtual status_t set_default_params() override {
+            if (src_pd_.desc()->format == memory_format::any)
```
Comment (emfomenk) [inline src/cpu/jit_uni_pooling.hpp]: The MKL-DNN logic is that only compute-bound primitives (like (de-)convolution and inner-product) may define the formats they want. All the other primitives (like relu, pulling) should follow whatever a user provides them.
So this line should not appear here...

Comment (tkarna) [reply src/cpu/jit_uni_pooling.hpp]: I see, I'll remove that.

Comment (tkarna) [reply src/cpu/jit_uni_pooling.hpp]: Done.

**Comment by emfomenk** [comment]
Thanks!

**Pull request merged**